### PR TITLE
Refactor rnp_key_get_default_key() to avoid code duplication.

### LINKED ISSUE #1570: Refactor rnp_key_get_default_key() to avoid code duplication.
## Description

Currently it duplicates some code from the `rnp_locate_key_int()`. This could be addressed by adding `require_secret` parameter to the later one, so no extra allocation/deallocation would happen. More details: https://github.com/rnpgp/rnp/pull/1557#discussion_r675110135

### PR DESCRIPTION
**Pull request by antonsviridenko**
closes #1570

### CODE REVIEW

File: src/lib/rnp.cpp [hunk 1]
```diff
@@ -3546,13 +3546,21 @@ locator_to_str(const pgp_key_search_t *locator,
 }
 
 static rnp_result_t
-rnp_locate_key_int(rnp_ffi_t ffi, const pgp_key_search_t &locator, rnp_key_handle_t *handle)
+rnp_locate_key_int(rnp_ffi_t               ffi,
+                   const pgp_key_search_t &locator,
+                   rnp_key_handle_t *      handle,
+                   bool                    require_secret)
```
Comment (ni4) [inline src/lib/rnp.cpp:3552]: What about `bool require_secret = false`, as it is true only in one call?

File: src/lib/rnp.cpp [hunk 2]
```diff
@@ -6207,33 +6216,21 @@ try {
         *default_key = NULL;
         return RNP_ERROR_NO_SUITABLE_KEY;
     }
+    if (keyflag != PGP_KF_ENCRYPT) {
```
Comment (ni4) [inline src/lib/rnp.cpp:6219]: `bool require_secret = keyflag != PGP_KF_ENCRYPT` for less code lines?

File: src/lib/rnp.cpp [hunk 3]
```diff
@@ -6207,33 +6216,21 @@ try {
...
     pgp_key_search_t search = {(pgp_key_search_type_t) 0};
     search.type = PGP_KEY_SEARCH_FINGERPRINT;
     search.by.fingerprint = defkey->fp();
 
-    // search pubring
-    pgp_key_t *pub = rnp_key_store_search(primary_key->ffi->pubring, &search, NULL);
-    // search secring
-    pgp_key_t *sec = rnp_key_store_search(primary_key->ffi->secring, &search, NULL);
+    rnp_result_t ret;
+    ret = rnp_locate_key_int(primary_key->ffi, search, default_key, require_secret);
```
Comment (ni4) [inline src/lib/rnp.cpp:6227]: `rnp_result_t ret = rnp_locate_key_int(...)` for less code lines?

File: src/lib/rnp.cpp [hunk 4]
```diff
@@ -6207,33 +6216,21 @@ try {
...
 
-    if (!sec && keyflag != PGP_KF_ENCRYPT) {
+    if (*default_key == NULL && ret == RNP_SUCCESS) {
```
Comment (ni4) [inline src/lib/rnp.cpp:6229]: We are more stick to the `if (!*default_key && !ret)` convention.

**Review by ni4 — CHANGES_REQUESTED** [review]
LGTM, just some coding style comments.

**Comment by antonsviridenko** [comment]
@ni4 done

**Review by ni4 — APPROVED** [review]
LGTM, thanks, @antonsviridenko !

**Review by ronaldtse — APPROVED** [review]
Thanks @ni4 !

**Pull request merged**